core/src/calleeFunctions/helpers/oneInch.ts

```diff
@@ -0,0 +1,197 @@
+    if (functionSignature !== EXPECTED_SIGNATURE) {
+        throw new Error(`Unexpected 1inch function signature: ${functionSignature}, expected: ${EXPECTED_SIGNATURE}`);
+    }
+    return oneinchResponse;
+}
+
+export async function extractPathFromSwapResponseProtocols(
+    network: string,
+    oneInchRoutes: OneInchSwapRoute[]
+): Promise<string[]> {
+    const pathStepsResolves = await Promise.all(
+        oneInchRoutes[0].map(async route => {
+            return await Promise.all([
+                await getContractSymbolByAddress(network, route[0].fromTokenAddress),
+                await getContractSymbolByAddress(network, route[0].toTokenAddress),
+            ]);
+        })
+    );
+    const path = [pathStepsResolves[0][0]];
+    for (const step of pathStepsResolves) {
+        path.push(step[1]);
+    }
+    return path;
+}
+
+export async function getOneInchQuote(network: string, collateralSymbol: string, amount: string, marketId: string) {
+    let chainId = parseInt(getChainIdByNetworkType(network) || '', 16);
+    if (chainId === 1337) {
+        // TODO: remove this hack
+        chainId = 1;
+    }
+    if (Number.isNaN(chainId)) {
+        throw new Error(`Invalid chainId: ${chainId}`);
+    }
+    const toTokenAddress = await getTokenAddressByNetworkAndSymbol(network, 'DAI');
+    const fromTokenAddress = await getTokenAddressByNetworkAndSymbol(network, collateralSymbol);
+    const calleeAddress = getCalleeAddressByCollateralType(
+        network,
+        getCollateralConfigBySymbol(collateralSymbol).ilk,
+        marketId
+    );
+    // Documentation https://docs.1inch.io/docs/aggregation-protocol/api/swap-params/
+    const swapParams = {
+        fromTokenAddress,
+        toTokenAddress,
+        fromAddress: calleeAddress,
+        amount,
+        protocols: (await getOneinchValidProtocols(chainId)).join(','),
+    };
+    const oneinchResponse = await executeOneInchApiRequest(chainId, '/quote', swapParams);
+    return {
+        estimatedGas: oneinchResponse?.estimatedGas,
+        route: await extractPathFromSwapResponseProtocols(network, oneinchResponse?.protocols),
+        tokenOut: oneinchResponse?.toTokenAmount,
+    };
+}
+
+export async function getOneInchMarketData(
+    network: string,
+    collateral: CollateralConfig,
+    amount: BigNumber,
+    marketId: string
+) {
+    const swapData = await getOneinchSwapParameters(
+        network,
+        collateral.symbol,
+        amount.shiftedBy(WAD_NUMBER_OF_DIGITS).toFixed(0),
+        marketId
+    );
+    const path = await extractPathFromSwapResponseProtocols(network, swapData.protocols);
+    const calleeData = swapData.tx.data;
+    const estimatedGas = new BigNumber(
+        (
+            await getOneInchQuote(
+                network,
+                collateral.symbol,
+                amount.shiftedBy(WAD_NUMBER_OF_DIGITS).toFixed(0),
+                marketId
+            )
+        ).estimatedGas
+    );
```

Comment:
Please don't do another request to 1inch just to estimate gas of the transaction. `swapData.tx.gasPrice` already contains enough data to calculate transaction cost